controllers/etcdcopybackupstask/reconciler_test.go

```diff
@@ -132,10 +139,525 @@ var _ = Describe("EtcdCopyBackupsTaskController", func() {
+			druidutils.ABS,
+			druidutils.GCS,
+			druidutils.S3,
+			druidutils.Swift,
+			druidutils.OSS,
+			druidutils.OCS,
+		} {
+			Context(fmt.Sprintf("with provider #%s", p), func() {
+				provider := p
+				BeforeEach(func() {
+					storageProvider := druidv1alpha1.StorageProvider(provider)
+					storeSpec = &druidv1alpha1.StoreSpec{
+						Container: &container,
+						Provider:  &storageProvider,
+					}
+				})
+
+				It("should create the correct env vars", func() {
+					envVars := createEnvVarsFromStore(storeSpec, provider, envKeyPrefix, volumePrefix)
+					checkEnvVars(envVars, provider, container, envKeyPrefix, volumePrefix)
+
+				})
+			})
+		}
+		Context("with provider #Local", func() {
+			BeforeEach(func() {
+				storageProvider := druidv1alpha1.StorageProvider(druidutils.Local)
+				storeSpec = &druidv1alpha1.StoreSpec{
+					Container: &container,
+					Provider:  &storageProvider,
+				}
+			})
+
+			It("should create the correct env vars", func() {
+				envVars := createEnvVarsFromStore(storeSpec, druidutils.Local, envKeyPrefix, volumePrefix)
+				checkEnvVars(envVars, druidutils.Local, container, envKeyPrefix, volumePrefix)
+
+			})
+		})
+	})
+
+	Describe("#createVolumeMountsFromStore", func() {
+		var (
+			volumeMountPrefix = "source-"
+			storeSpec         *druidv1alpha1.StoreSpec
+		)
+		// Loop through different storage providers to test with
+		for _, p := range []string{
+			druidutils.Local,
+			druidutils.ABS,
+			druidutils.GCS,
+			druidutils.S3,
+			druidutils.Swift,
+			druidutils.OSS,
+			druidutils.OCS,
+		} {
+			Context(fmt.Sprintf("with provider #%s", p), func() {
+				provider := p
+				BeforeEach(func() {
+					storageProvider := druidv1alpha1.StorageProvider(provider)
+					storeSpec = &druidv1alpha1.StoreSpec{
+						Container: pointer.String("source-container"),
+						Provider:  &storageProvider,
+					}
+				})
+
+				It("should create the correct volume mounts", func() {
+					volumeMounts := createVolumeMountsFromStore(storeSpec, provider, volumeMountPrefix)
+					Expect(volumeMounts).To(HaveLen(1))
+
+					expectedMountPath := ""
+					expectedMountName := ""
+
+					switch provider {
+					case druidutils.Local:
+						expectedMountName = volumeMountPrefix + "host-storage"
+						expectedMountPath = *storeSpec.Container
+					case druidutils.GCS:
+						expectedMountName = volumeMountPrefix + "etcd-backup"
+						expectedMountPath = "/root/." + volumeMountPrefix + "gcp/"
+					case druidutils.S3, druidutils.ABS, druidutils.Swift, druidutils.OCS, druidutils.OSS:
+						expectedMountName = volumeMountPrefix + "etcd-backup"
+						expectedMountPath = "/root/" + volumeMountPrefix + "etcd-backup/"
+					default:
+						Fail(fmt.Sprintf("Unknown provider: %s", provider))
+					}
+
+					Expect(volumeMounts[0].Name).To(Equal(expectedMountName))
+					Expect(volumeMounts[0].MountPath).To(Equal(expectedMountPath))
+				})
+			})
+		}
+
+		Context("with provider #Local", func() {
```

Comment:
you have already included provider local in the test case above. Any reason to create another test again for local provider?